### PR TITLE
Remove TICL dedicated wfs, and adjust the name for the pfTICLProducer automatic config

### PR DESCRIPTION
**Pull request by perrotta**
This is a fix proposed to solve the issue #29446 (credit to @rovere for the suggestion)

The failing workflows are the ones that were customized to run TICL. Now that TICL is part of the official reconstruction, there's no need to have them as separate workflows.
This PR removes the wf 20493.52 and its customization from runTheMatrix

In addition, the config automatically created with fillDescriptions for the PFTiclProducer producer is here renamed according to the name of the producer itself.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
please test

**Comment by rovere** [comment]
@perrotta 
As explained in #29446, I don't think this proposed solution is acceptable.
I'd close this PR.

**Comment by perrotta** [comment]
please abort

**Comment by perrotta** [comment]
@rovere I still believe that calling the automatic config created by fillDescription() with the same name of the producer itself could have been more reasonable rather than giving it a different name, even if this is not the actual fix for the issue pointed out,

But I'd let you guys decide about HGCal stuffs, and I won't insist on it if you don't think so.

Then, please confirm that you still suggest to close this one, and I will

**Comment by rovere** [comment]
@perrotta I think we are discussing 2 different topics.

* If this PR was meant to fix the crash of the offending workflows as reported in #29446, then I'd argue, as I said, that the proposed solution is not the optimal one.

* If instead, you want to have symmetry between the `C++` class name and the `python` label used within the `fillDescription` call, I won't object.

I can propose, at this point, to extend the scope of this PR and include also the remove the offending workflows from runTheMatrix.

You can find a possible patch [here](https://rovere.web.cern.ch/rovere/HGCAL/RemoveTICLWorkflows.patch).

**Comment by perrotta** [comment]
Yes, sure. After your explanation this PR was just meant to resolve the symmetry, as you say.

I can add the commit you point out and concentrate here also the fix.
Further to it, we should also explicitely remove the offending workflow, i.e. remove the line 
https://cmssdt.cern.ch/lxr/source/Configuration/PyReleaseValidation/python/relval_2026.py#0022

**Comment by perrotta** [comment]
Thank you @rovere 

Please confirm that the last commit does correspond to what you propose.
I will launch the test (and update the PR description) if so,

Otherwise, just feel free of submitting a patch yourself, if you think it can be faster.

**Comment by rovere** [comment]
Thanks to you, @perrotta for the very fast reaction.
I'm fine with the proposed changes.

**Comment by perrotta** [comment]
please test

File: RecoHGCal/TICL/python/iterativeTICL_cff.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from RecoHGCal.TICL.HADStep_cff import *
 from RecoHGCal.TICL.ticlLayerTileProducer_cfi import ticlLayerTileProducer
 from RecoHGCal.TICL.ticlCandidateFromTrackstersProducer_cfi import ticlCandidateFromTrackstersProducer as _ticlCandidateFromTrackstersProducer
-from RecoHGCal.TICL.pfTICL_cfi import pfTICL
+from RecoHGCal.TICL.pfTICLProducer_cfi import pfTICLProducer as pfTICL
```
Comment (slava77) [inline RecoHGCal/TICL/python/iterativeTICL_cff.py:9]: This is something I am a bit suspicious about, but never checked explicitly.
If somebody else imports ```pfTICLProducer``` as is (```pfTICLProducer```) and modifies it, would this modification apply to what's called ```pfTICL``` here?
If that's the case, I'd rather we have an explicit clone here (like it's done for all other "import x as y".

Comment (perrotta) [reply RecoHGCal/TICL/python/iterativeTICL_cff.py]: good point @slava77 
clone() added now

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- It fixes the building error in wf 20493.52 by removing it from the matrix
- Jenkins tests pass and show no differences

File: RecoHGCal/TICL/python/iterativeTICL_cff.py (hunk 1 above)
Comment (makortel) [reply RecoHGCal/TICL/python/iterativeTICL_cff.py]: Even though it was changed to a `clone()` already, let me clarify.

> This is something I am a bit suspicious about, but never checked explicitly.
> If somebody else imports `pfTICLProducer` as is (`pfTICLProducer`) and modifies it, would this modification apply to what's called `pfTICL` here?

Yes.

> If that's the case, I'd rather we have an explicit clone here (like it's done for all other "import x as y".

The core group recommends to modify configuration objects only in the cfi/cff files the objects are defined (a clone is counted as defining a new object, and this includes modifications with `cms.Modifier`s).

**Comment by kpedro88** [comment]
+upgrade

**Comment by silviodonato** [comment]
I will integrate this PR in the next IB, if @chayanit @pgunnell @wajidalikhan don't have any objection.

**Comment by chayanit** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**